aptos-move/framework/aptos-framework/sources/object.move

```diff
@@ -464,10 +470,15 @@ module aptos_framework::object {
         let current_address = object.owner;
 
         let count = 0;
-        while (owner != current_address) {
+        while (owner != current_address)
+        {
             let count = count + 1;
             assert!(count < MAXIMUM_OBJECT_NESTING, error::out_of_range(EMAXIMUM_NESTING));
 
+            spec {
+                assume destination == current_address;
```

Comment:
I am not totally convinced that we should use assume here. Are you using it to define aborts_if condition?